app/src/main/java/com/giraffe/cineverseapp/main_activity/MainViewModel.kt

```diff
@@ -0,0 +1,69 @@
+package com.giraffe.cineverseapp.main_activity
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.giraffe.user.usecase.IsLoggedInUseCase
+import com.giraffe.user.usecase.IsOnBoardingFirstTimeUseCase
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class MainViewModel @Inject constructor(
+    private val isLoggedInUseCase: IsLoggedInUseCase,
+    private val isOnBoardingFirstTimeUseCase: IsOnBoardingFirstTimeUseCase
+) : ViewModel() {
+    private val _state = MutableStateFlow(MainUiState())
+    val state: StateFlow<MainUiState> = _state.asStateFlow()
+
+    init {
+        checkIfOnBoardingFirstTime()
+        checkIsLoggedIn()
+    }
+
+
+    fun checkIfOnBoardingFirstTime() {
+        safeExecute(
+            onSuccess = { result ->
+                _state.value = _state.value.copy(
+                    isOnBoardingFirstTime = result,
+                )
+            }
```

Comment:
you can create a method for success case like:

```suggestion
              onSuccess = ::onCheckOnBoardingFirstTimeSuccess
```